src/groovy/com/unifina/domain/CustomStreamIDValidator.groovy

```diff
@@ -0,0 +1,23 @@
+package com.unifina.domain
+
+import grails.compiler.GrailsCompileStatic
+
+@GrailsCompileStatic
+class CustomStreamIDValidator {
```

Comment:
We tend to prefer Java classes unless there's a reason to use Groovy (Grails artifacts etc., or some particular thing is much easier to write). In this class it doesn't seem to add value IMO.